File: python/CustomQueue.py
class CustomQueue:
    def __init__(self):
        self.items = []
    
    def enqueue(self, item):
        self.items.append(item) # type: ignore
    
    def dequeue(self):
        return self.items.pop(0) if self.items else None

    def peek(self):
        return self.items[0] if self.items else None

queue = CustomQueue()

queue.enqueue(1)
queue.enqueue(2)
queue.enqueue(3)

print(queue.peek())

File: java/CustomQueue.java
import java.util.LinkedList;
import java.util.Queue;

public class CustomQueue {

    private Queue<Integer> items;

    public CustomQueue() {
        this.items = new LinkedList<>();
    }

    public void enqueue(int item) {
        this.items.add(item);
    }

    public int dequeue() {
        return this.items.poll();
    }

    public int peek() {
        return this.items.peek();
    }

    public static void main(String[] args) {
        CustomQueue queue = new CustomQueue();

        queue.enqueue(0);
        queue.dequeue();
        queue.enqueue(1);
        queue.enqueue(2);

        System.out.println(queue.peek());
    }
}
